blackbox/docs/src/doc_tests/tests.py

```diff
@@ -431,115 +350,162 @@ def setUpTutorials(test):
+                            'general/builtins/aggregation.rst',
+                            'general/builtins/arithmetic.rst',
+                            'general/builtins/scalar.rst',
+                            'general/builtins/table-functions.rst',
+                            'admin/system-information.rst',
+                            'general/dql/selects.rst',
+                            'interfaces/postgres.rst',):
+        s = docsuite(fn,
+                     setUp=setUpLocationsAndQuotes)
         s.layer = crate_layer
-        docs_suite.addTest(s)
-    for fn in ('general/dql/joins.rst',
-               'general/builtins/subquery-expressions.rst',):
-        path = os.path.join('..', '..', fn)
-        s = docsuite(path,
-                     parser=crash_parser,
-                     setUp=setUpEmpDeptAndColourArticlesAndGeo,
-                     tearDown=tearDownEmpDeptAndColourArticlesAndGeo)
+        tests.append(s)
+
+    for fn in doctest_files('general/dql/geo.rst',):
+        s = docsuite(fn,
+                     setUp=setUpCountries)
         s.layer = crate_layer
-        docs_suite.addTest(s)
-    for fn in ('general/dml.rst',):
-        s = docsuite('../../' + fn,
-                     parser=crash_parser,
-                     setUp=setUpLocationsQuotesAndUserVisits,
-                     tearDown=tearDownLocationsQuotesAndUserVisits)
+        tests.append(s)
+
+    for fn in doctest_files('general/dql/joins.rst',
+                            'general/builtins/subquery-expressions.rst',):
+        s = docsuite(fn,
+                     setUp=setUpEmpDeptAndColourArticlesAndGeo)
         s.layer = crate_layer
-        docs_suite.addTest(s)
-    for fn in ('admin/snapshots.rst',):
-        s = docsuite('../../' + fn,
-                     parser=crash_parser,
-                     setUp=setUpLocationsAndQuotes,
-                     tearDown=tearDownLocationsAndQuotes)
+        tests.append(s)
+
+    for fn in doctest_files('general/dml.rst',):
+        s = docsuite(fn,
+                     setUp=setUpLocationsQuotesAndUserVisits)
         s.layer = crate_layer
-        docs_suite.addTest(s)
-    for fn in ('general/dql/union.rst',):
-        path = os.path.join('..', '..', fn)
-        s = doctest.DocFileSuite(path,
-                                 parser=crash_parser,
-                                 setUp=setUpPhotosAndCountries,
-                                 tearDown=tearDownPhotosAndCountries,
-                                 optionflags=doctest.NORMALIZE_WHITESPACE |
-                                             doctest.ELLIPSIS,
-                                 encoding='utf-8')
+        tests.append(s)
+
+    for fn in doctest_files('general/dql/union.rst',):
+        s = docsuite(fn,
+                     setUp=setUpPhotosAndCountries)
         s.layer = crate_layer
-        docs_suite.addTest(s)
+        tests.append(s)
+
+    random.shuffle(tests)
```

Comment:
Maybe add a short comment that we do this to avoid the tests becoming dependent on each other